### PR TITLE
test : fix test data race

### PR DESCRIPTION
**Pull request by lhy1024**
Signed-off-by: lhy1024 <admin@liudos.us>

### What problem does this PR solve? 

split  test to avoid data race

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by nolouch** [comment]
where is the race record?

**Comment by rleungx** [comment]
I think this PR is trying to fix this race problem. @nolouch 
```
==================
WARNING: DATA RACE
Read at 0x00c000ba98a0 by goroutine 979:
  github.com/pingcap/pd/server/config.(*ScheduleOption).GetLeaderScheduleLimit()
      /home/ryan/pd/server/config/option.go:190 +0xf1
  github.com/pingcap/pd/server.(*namespaceCluster).GetLeaderScheduleLimit()
      /home/ryan/pd/server/namespace_cluster.go:146 +0xc0
  github.com/pingcap/pd/server/schedulers.(*balanceHotRegionsScheduler).allowBalanceLeader()
      /home/ryan/pd/server/schedulers/hot_region.go:145 +0x14e
  github.com/pingcap/pd/server/schedulers.(*balanceHotRegionsScheduler).balanceByLeader()
      /home/ryan/pd/server/schedulers/hot_region.go:356 +0x6d
  github.com/pingcap/pd/server/schedulers.(*balanceHotRegionsScheduler).balanceHotWriteRegions()
      /home/ryan/pd/server/schedulers/hot_region.go:219 +0x14d
  github.com/pingcap/pd/server/schedulers.(*balanceHotRegionsScheduler).dispatch()
      /home/ryan/pd/server/schedulers/hot_region.go:167 +0x3ca
  github.com/pingcap/pd/server/schedulers.(*balanceHotRegionsScheduler).Schedule()
      /home/ryan/pd/server/schedulers/hot_region.go:154 +0x1f2
  github.com/pingcap/pd/server.scheduleByNamespace()
      /home/ryan/pd/server/namespace_cluster.go:138 +0x1dc
  github.com/pingcap/pd/server.(*scheduleController).Schedule()
      /home/ryan/pd/server/coordinator.go:499 +0x136
  github.com/pingcap/pd/server.(*coordinator).runScheduler()
      /home/ryan/pd/server/coordinator.go:450 +0x402

Previous write at 0x00c000ba98a0 by goroutine 619:
  github.com/pingcap/pd/server.(*testCoordinatorSuite).TestCheckRegion()
      /home/ryan/pd/server/coordinator_test.go:371 +0x13c3
  fmt.(*ss).doScanf()
      /usr/local/go/src/fmt/scan.go:1217 +0x382
  fmt.Fscanf()
      /usr/local/go/src/fmt/scan.go:143 +0xcf
  github.com/syndtr/goleveldb/leveldb/storage.fsParseName()
      /usr/local/go/src/fmt/scan.go:114 +0x192
  github.com/syndtr/goleveldb/leveldb/storage.(*fileStorage).List()
      /home/ryan/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/storage/file_storage.go:458 +0x2c4
  fmt.Fscanf()
      /usr/local/go/src/fmt/scan.go:143 +0xcf
  github.com/syndtr/goleveldb/leveldb/storage.fsParseName()
      /usr/local/go/src/fmt/scan.go:114 +0x192
  github.com/syndtr/goleveldb/leveldb/storage.(*fileStorage).List()
      /home/ryan/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/storage/file_storage.go:458 +0x2c4
  github.com/syndtr/goleveldb/leveldb.(*DB).checkAndCleanFiles()
      /home/ryan/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/db_util.go:52 +0x2f8
  github.com/syndtr/goleveldb/leveldb.openDB()
      /home/ryan/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/db.go:130 +0x9f0
  github.com/syndtr/goleveldb/leveldb/storage.(*fileStorage).List()
      /home/ryan/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/storage/file_storage.go:458 +0x2c4
  fmt.Fscanf()
      /usr/local/go/src/fmt/scan.go:143 +0xcf
  github.com/syndtr/goleveldb/leveldb/storage.fsParseName()
      /usr/local/go/src/fmt/scan.go:114 +0x192
  github.com/syndtr/goleveldb/leveldb/storage.(*fileStorage).List()
      /home/ryan/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/storage/file_storage.go:458 +0x2c4
  fmt.Fscanf()
      /usr/local/go/src/fmt/scan.go:143 +0xcf
  github.com/syndtr/goleveldb/leveldb/storage.fsParseName()
      /usr/local/go/src/fmt/scan.go:114 +0x284
  fmt.(*ss).doScanf()
      /usr/local/go/src/fmt/scan.go:1217 +0x382
  fmt.Fscanf()
      /usr/local/go/src/fmt/scan.go:143 +0xcf
  github.com/syndtr/goleveldb/leveldb/storage.fsParseName()
      /usr/local/go/src/fmt/scan.go:114 +0x192
  github.com/syndtr/goleveldb/leveldb/storage.(*fileStorage).List()
      /home/ryan/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/storage/file_storage.go:458 +0x2c4
  fmt.Fscanf()
      /usr/local/go/src/fmt/scan.go:143 +0xcf
  github.com/syndtr/goleveldb/leveldb/storage.fsParseName()
      /usr/local/go/src/fmt/scan.go:114 +0x192
  github.com/syndtr/goleveldb/leveldb/storage.(*fileStorage).List()
      /home/ryan/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/storage/file_storage.go:458 +0x2c4
  github.com/syndtr/goleveldb/leveldb.(*DB).recoverJournal()
      /home/ryan/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/db.go:476 +0xc5
  github.com/syndtr/goleveldb/leveldb.openDB()
      /home/ryan/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/db.go:125 +0x9cc
  github.com/syndtr/goleveldb/leveldb.Open()
      /home/ryan/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/db.go:197 +0x1c6
  fmt.Fscanf()
      /usr/local/go/src/fmt/scan.go:143 +0xcf
  github.com/syndtr/goleveldb/leveldb/storage.fsParseName()
      /usr/local/go/src/fmt/scan.go:114 +0x192
  github.com/syndtr/goleveldb/leveldb/storage.(*fileStorage).List()
      /home/ryan/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/storage/file_storage.go:458 +0x2c4
  github.com/syndtr/goleveldb/leveldb.(*session).recover.func1()
      /home/ryan/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/session.go:113 +0xf4
  github.com/syndtr/goleveldb/leveldb.(*session).recover()
      /home/ryan/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/session.go:121 +0x1929
  github.com/syndtr/goleveldb/leveldb.Open()
      /home/ryan/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/db.go:183 +0xf8
  github.com/syndtr/goleveldb/leveldb.OpenFile()
      /home/ryan/go/pkg/mod/github.com/syndtr/goleveldb@v0.0.0-20180815032940-ae2bd5eed72d/leveldb/db.go:219 +0xa5
  github.com/pingcap/pd/server/kv.NewLeveldbKV()
      /home/ryan/pd/server/kv/levedb_kv.go:31 +0x52
  github.com/pingcap/pd/server/core.NewRegionStorage()
      /home/ryan/pd/server/core/region_storage.go:53 +0x5d
  github.com/pingcap/pd/server.getHeartBeatStreams()
      /home/ryan/pd/server/coordinator_test.go:1243 +0x289
  github.com/pingcap/pd/server.(*testCoordinatorSuite).TestCheckRegion()
      /home/ryan/pd/server/coordinator_test.go:310 +0x10f
  runtime.call32()
      /usr/local/go/src/runtime/asm_amd64.s:539 +0x3a
  reflect.Value.Call()
      /usr/local/go/src/reflect/value.go:321 +0xd3
  github.com/pingcap/check.(*suiteRunner).forkTest.func1()
      /home/ryan/go/pkg/mod/github.com/pingcap/check@v0.0.0-20190102082844-67f458068fc8/check.go:836 +0x9aa
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /home/ryan/go/pkg/mod/github.com/pingcap/check@v0.0.0-20190102082844-67f458068fc8/check.go:730 +0xc4
```

File: server/coordinator_test.go [hunk 1]
```diff
@@ -315,36 +315,40 @@ func MaxUint64(nums ...uint64) uint64 {
 	return result
 }
 
-func (s *testCoordinatorSuite) TestCheckRegion(c *C) {
-	_, opt, err := newTestScheduleConfig()
-	c.Assert(err, IsNil)
+func (s *testCoordinatorSuite) prepare(cfg *config.ScheduleConfig, opt *config.ScheduleOption, c *C) (*testCluster, *heartbeatStreams, func(), *coordinator) {
```
Comment (rleungx) [inline server/coordinator_test.go]: It's better to use it for other tests in the file?

File: server/coordinator_test.go [hunk 2]
```diff
@@ -368,21 +372,33 @@ func (s *testCoordinatorSuite) TestCheckRegion(c *C) {
...
+	tc, hbStreams, cleanup, co := s.prepare(cfg, opt, c)
+	defer cleanup()
+	defer hbStreams.Close()
 
-	//test checkerIsBusy
-	co.cluster.opt.Load().ReplicaScheduleLimit = 10
-	co.cluster.opt.Load().LeaderScheduleLimit = 10
-	co.cluster.opt.Load().RegionScheduleLimit = 10
-	co.cluster.opt.Load().MergeScheduleLimit = 10
+	storeNum := uint64(4)
```
Comment (rleungx) [inline server/coordinator_test.go]: `var storeNum uint64 = 1`?

**Comment by rleungx** [comment]
BTW, It seems that the log in line 406 is not useful?

File: server/coordinator_test.go (hunk 1 above)
Comment (lhy1024) [reply server/coordinator_test.go]: Good point. I use some and find there are many changes, I will push them in new PR.

**Comment by lhy1024** [comment]
@rleungx PTAL

**Comment by lhy1024** [comment]
> BTW, It seems that the log in line 406 is not useful?

@rleungx I have to debug the extra long log problem and I removed it now.

**Comment by disksing** [comment]
PTAL @rleungx

**Comment by lhy1024** [comment]
PTAL @rleungx

**Pull request merged**